File: evaluations/matpower/verify_install.m
% Verify MATPOWER installation by running power flow on IEEE 39-bus case.
%
% Usage: octave verify_install.m

% Add MATPOWER to path
addpath(fullfile(pwd, 'matpower-8.1', 'lib'));
addpath(fullfile(pwd, 'matpower-8.1', 'data'));

% Load IEEE 39-bus case from shared data directory
mpc = loadcase(fullfile('..', '..', 'data', 'networks', 'case39.m'));

fprintf('MATPOWER version: %s\n', mpver());
fprintf('Buses: %d\n', size(mpc.bus, 1));
fprintf('Branches: %d\n', size(mpc.branch, 1));

% Run DC power flow
results = rundcpf(mpc);

if results.success
    fprintf('DC power flow completed successfully\n');
    exit(0);
else
    fprintf('DC power flow failed\n');
    exit(1);
end

File: evaluations/pandapower/verify_install.py
"""Verify pandapower installation by running DC power flow on IEEE 39-bus case."""

from __future__ import annotations

import sys
from pathlib import Path

import pandapower as pp

DATA_DIR = Path(__file__).resolve().parent.parent.parent / "data" / "networks"

# Load case39.m directly
net = pp.converter.from_mpc(str(DATA_DIR / "case39.m"))

# Run DC power flow
pp.rundcpp(net)

print(f"pandapower version: {pp.__version__}")
print(f"Buses: {len(net.bus)}")
print(f"Lines: {len(net.line)}")
print(f"Converged: {net.converged}")
sys.exit(0 if net.converged else 1)
